### PR TITLE
Fix stubtest errors for updating aiofiles.

### PR DESCRIPTION
**Pull request by hmc-cs-mdrissi**
Resolves [stubsabot](https://github.com/python/typeshed/pull/9703). Not a library I've used so generally read over implementation.

### CODE REVIEW

File: stubs/aiofiles/@tests/stubtest_allowlist.txt [hunk 1]
```diff
@@ -9,6 +9,15 @@ aiofiles.threadpool.binary.AsyncBufferedIOBase.readable
 aiofiles.threadpool.binary.AsyncBufferedIOBase.seekable
 aiofiles.threadpool.binary.AsyncBufferedIOBase.tell
 aiofiles.threadpool.binary.AsyncBufferedIOBase.writable
+aiofiles.threadpool.binary.AsyncIndirectBufferedIOBase.close
```
Comment (hmc-cs-mdrissi) [inline stubs/aiofiles/@tests/stubtest_allowlist.txt:12]: All of the allowlist are copy pasting the non-indirect version above it. This version introduces indirect variants of most of IO classes that need exception for same reason (at runtime they delegate with *args/**kwargs).

File: stubs/aiofiles/aiofiles/base.pyi [hunk 2]
```diff
@@ -13,6 +13,9 @@ class AsyncBase(Generic[_T]):
     def __aiter__(self: Self) -> Self: ...
     async def __anext__(self) -> _T: ...
 
+class AsyncIndirectBase(AsyncBase[_T]):
+    def __init__(self, name: str, loop: Any, executor: Any, indirect: Callable[[], TextIO | BinaryIO]) -> None: ...
```
Comment (hmc-cs-mdrissi) [inline stubs/aiofiles/aiofiles/base.pyi:17]: I'm least confident about the type for indirect. In implementation indirect is only passed sys.stdin/sys.stdin.buffer/etc. It looks like intent is if _T is str then TextIO (or maybe IO[str]) while if _T is bytes then BinaryIO. 

Could add overloads to describe _T relationship although confused why _T is unconstrained type variable (seems like it should be str/bytes always).

File: stubs/aiofiles/aiofiles/threadpool/binary.pyi [hunk 3]
```diff
@@ -34,8 +36,15 @@ class AsyncBufferedIOBase(_UnknownAsyncBinaryIO):
...
 class AsyncBufferedReader(AsyncBufferedIOBase):
     async def peek(self, __size: int = ...) -> bytes: ...
 
+class AsyncIndirectBufferedReader(AsyncIndirectBase[bytes], AsyncBufferedIOBase):
+    async def peek(self, __size: int = ...) -> bytes: ...
+
 class AsyncFileIO(_UnknownAsyncBinaryIO):
     async def readall(self) -> bytes: ...
+
+class AsyncIndirectFileIO(AsyncIndirectBase[bytes], AsyncFileIO): ...
```
Comment (AlexWaygood) [inline stubs/aiofiles/aiofiles/threadpool/binary.pyi:50]: Similar comment to the one I made about `AsyncIndirectBufferedIOBase` -- I'm wary of inheriting from `AsyncFileIO` here.

File: stubs/aiofiles/aiofiles/threadpool/text.pyi [hunk 4]
```diff
@@ -37,3 +37,5 @@ class AsyncTextIOWrapper(AsyncBase[str]):
     def name(self) -> FileDescriptorOrPath: ...
     @property
     def mode(self) -> str: ...
+
+class AsyncTextIndirectIOWrapper(AsyncIndirectBase[str], AsyncTextIOWrapper): ...
```
Comment (AlexWaygood) [inline stubs/aiofiles/aiofiles/threadpool/text.pyi:41]: Similar comment to the one I made about AsyncIndirectBufferedIOBase -- I'm wary of inheriting from `AsyncTextIOWrapper` here.

**Review by AlexWaygood — COMMENTED** [review]
Thank you, and sorry for the delay in anybody getting to this! A few thoughts:

**Comment by JelleZijlstra** [comment]
@hmc-cs-mdrissi could you respond to Alex's feedback?

File: stubs/aiofiles/aiofiles/threadpool/binary.pyi [hunk 5]
```diff
@@ -34,8 +36,15 @@ class AsyncBufferedIOBase(_UnknownAsyncBinaryIO):
     @property
     def raw(self) -> FileIO: ...
 
+class AsyncIndirectBufferedIOBase(AsyncIndirectBase[bytes], AsyncBufferedIOBase): ...
+
 class AsyncBufferedReader(AsyncBufferedIOBase):
     async def peek(self, __size: int = ...) -> bytes: ...
 
+class AsyncIndirectBufferedReader(AsyncIndirectBase[bytes], AsyncBufferedIOBase):
```
[the remark this answers is not in the document]
Comment (hmc-cs-mdrissi) [reply stubs/aiofiles/aiofiles/threadpool/binary.pyi:44]: Guessing you meant AsyncIndirectBufferedIOBase double checking the runtime?

File: stubs/aiofiles/aiofiles/threadpool/binary.pyi [hunk 6]
```diff
@@ -34,8 +36,15 @@ class AsyncBufferedIOBase(_UnknownAsyncBinaryIO):
     @property
     def raw(self) -> FileIO: ...
 
+class AsyncIndirectBufferedIOBase(AsyncIndirectBase[bytes], AsyncBufferedIOBase): ...
```
[the remark this answers is not in the document]
Comment (hmc-cs-mdrissi) [reply stubs/aiofiles/aiofiles/threadpool/binary.pyi:39]: Sure I've duplicated methods needed from AsyncBufferedIOBase

File: stubs/aiofiles/aiofiles/threadpool/text.pyi (hunk 4 above)
Comment (hmc-cs-mdrissi) [reply stubs/aiofiles/aiofiles/threadpool/text.pyi]: This one inherits a lot of methods. I added _UnknownAsyncTextIO to have methods defined once and have both public text io's inherit from that. The mro now for public classes matches runtime.

If you'd prefer to not use that solution I'll do duplication way similar to others.

File: stubs/aiofiles/aiofiles/base.pyi (hunk 2 above)
Comment (hmc-cs-mdrissi) [reply stubs/aiofiles/aiofiles/base.pyi:17]: Hmm, I'd prefer to leave that for a follow up/future pr to be able to see effect by itself.

**Comment by hmc-cs-mdrissi** [comment]
Sorry on delay. I think I've covered all comments now.

File: stubs/aiofiles/aiofiles/base.pyi (hunk 2 above)
Comment (AlexWaygood) [reply stubs/aiofiles/aiofiles/base.pyi:17]: sure

File: stubs/aiofiles/aiofiles/threadpool/text.pyi (hunk 4 above)
Comment (AlexWaygood) [reply stubs/aiofiles/aiofiles/threadpool/text.pyi]: The solution you have looks good

**Review by AlexWaygood — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**